AzureFunctions.v3.DurableEntities/Orchestrator.cs

```diff
@@ -17,20 +16,22 @@ public static class Orchestrator
         public static async System.Threading.Tasks.Task<HttpResponseMessage> RunAsync([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req, [DurableClient] IDurableClient starter, ILogger log)
         {
             var reader = await req.Content.ReadAsStringAsync();
-            var events = EventGridEvent.ParseMany(BinaryData.FromString(reader));
-            var eventGridSoleItem = events?.SingleOrDefault();
-            if (eventGridSoleItem == null)
+            var events = EventGridEvent.Parse(BinaryData.FromString(reader));
+            if (events == null)
             {
                 return req.CreateResponse(HttpStatusCode.BadRequest, @"Expecting only one item in the Event Grid message");
             }
 
-            if (eventGridSoleItem.EventType == @"Microsoft.EventGrid.SubscriptionValidationEvent")
+            if (events.TryGetSystemEventData(out object eventData))
             {
-                log.LogTrace(@"Event Grid Validation event received.");
-                return req.CreateCompatibleResponse(HttpStatusCode.OK, $"{{ \"validationResponse\" : \"{((dynamic)JsonConvert.DeserializeObject(eventGridSoleItem.Data.ToString())).validationCode}\" }}");
+                if (eventData is SubscriptionValidationEventData subscriptionValidationEventData)
```

Comment:
Ah I see that the HttpRequestMessage API don't allow you to just pass a JSON object.
Could we do:
```cs
var evt = EventGridEvent.Parse(BinaryData.FromString(reader));
if (evt.EventType == "Microsoft.EventGrid.SubscriptionValidationEvent")
{
  log.LogTrace(@"Event Grid Validation event received.");
  return req.CreateCompatibleResponse(HttpStatusCode.OK, evt.Data.ToString());
}
```